src/views/RecordingPreviewer/RecordingPreviewerControls.tsx

```diff
@@ -0,0 +1,150 @@
+import { getRoutePath } from '@/routeMap'
+import { BrowserEvent } from '@/schemas/recording'
+import { useToast } from '@/store/ui/useToast'
+import { StudioFile } from '@/types'
+
+import { ExportScriptDialog } from '../Generator/ExportScriptDialog'
+
+interface RecordingPreviewControlsProps {
+  file: StudioFile
+  browserEvents: BrowserEvent[]
+}
+
+export function RecordingPreviewControls({
+  file,
+  browserEvents,
+}: RecordingPreviewControlsProps) {
+  const [showExportDialog, setShowExportDialog] = useState(false)
+  const showToast = useToast()
+  const navigate = useNavigate()
+  const createTestGenerator = useCreateGenerator()
+  const { fileName } = useParams()
+
+  // TODO: https://github.com/grafana/k6-studio/issues/277
+  // eslint-disable-next-line @typescript-eslint/no-unsafe-assignment
+  const { state } = useLocation()
+  // TODO: https://github.com/grafana/k6-studio/issues/277
+  // eslint-disable-next-line @typescript-eslint/no-unsafe-member-access
+  const isDiscardable = Boolean(state?.discardable)
+
+  const handleCreateGenerator = () => createTestGenerator(fileName)
+
+  const handleDeleteRecording = async () => {
+    await window.studio.ui.deleteFile(file)
+    navigate(getRoutePath('home'))
+  }
+
+  const handleDiscard = async () => {
+    await window.studio.ui.deleteFile(file)
+    navigate(getRoutePath('recorder'))
+  }
+
+  const handleExportBrowserScript = (fileName: string) => {
+    const test = convertToTest({
+      browserEvents,
+    })
+
+    emitScript(test)
+      .then((script) => window.studio.script.saveScript(script, fileName))
+      .then(() => {
+        navigate(
+          getRoutePath('validator', {
+            fileName: encodeURIComponent(fileName),
+          })
+        )
+      })
+      .catch((err) => {
+        console.error(err)
+
+        showToast({
+          title: 'Failed to export browser script.',
+          status: 'error',
+        })
+      })
+  }
+
+  return (
+    <>
+      {isDiscardable ? (
+        <Button onClick={handleDiscard} variant="outline" color="red">
+          Discard
+        </Button>
+      ) : (
+        <Button variant="outline" asChild>
+          <Link to={getRoutePath('recorder')}>New recording</Link>
+        </Button>
+      )}
+      <DropdownMenu.Root>
+        <DropdownMenu.Trigger>
+          <Button>
+            Create test <ChevronDownIcon />
+          </Button>
+        </DropdownMenu.Trigger>
+        <DropdownMenu.Content>
+          <MenuItem
+            label="HTTP test"
+            description="Set up a test generator to turn the recorded HTTP requests into a k6 script"
+            onClick={handleCreateGenerator}
+          />
+          <MenuItem
+            label="Browser test"
+            description="Export a k6 script simulating browser interactions"
+            disabled={browserEvents.length === 0}
+            onClick={() => setShowExportDialog(true)}
+          />
```

Comment:
Make sure to review the copy

Reply:
How about "Generate a script from HTTP requests using rules" or "Use rules to generate a script from HTTP requests"?